tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/GraphOperation.java

```diff
@@ -330,6 +384,165 @@ public int numControlConsumers() {
+   */
+  public float[] getAttrFloatList(String name) {
+    return getAttrFloatList(unsafeNativeHandle, name);
+  }
+
+  /**
+   * Get the value of a boolean attribute of this operation.
+   *
+   * @param name the name of the attribute
+   * @return the value of the attribute
+   */
+  public boolean getAttrBool(String name) {
+    return getAttrBool(unsafeNativeHandle, name);
+  }
+
+  /**
+   * Get the value of a boolean list attribute of this operation.
+   *
+   * @param name the name of the attribute
+   * @return the value of the attribute
+   */
+  public boolean[] getAttrBoolList(String name) {
+    return getAttrBoolList(unsafeNativeHandle, name);
+  }
+
+  /**
+   * Get the value of a data type attribute of this operation.
+   *
+   * @param name the name of the attribute
+   * @return the value of the attribute
+   */
+  public DataType getAttrType(String name) {
+    return getAttrType(unsafeNativeHandle, name);
+  }
+
+  /**
+   * Get the value of a data type list attribute of this operation.
+   *
+   * @param name the name of the attribute
+   * @return the value of the attribute
+   */
+  public DataType[] getAttrTypeList(String name) {
+    return getAttrTypeList(unsafeNativeHandle, name);
+  }
+
+  /**
+   * Get the value of a tensor attribute of this operation.
+   *
+   * @param name the name of the attribute
+   * @return the value of the attribute
+   */
+  public Tensor getAttrTensor(String name) {
+    return getAttrTensor(unsafeNativeHandle, name);
+  }
+
+  /**
+   * Get the value of a tensor list attribute of this operation.
+   *
+   * @param name the name of the attribute
+   * @return the value of the attribute
+   */
+  public Tensor[] getAttrTensorList(String name) {
+    return getAttrTensorList(unsafeNativeHandle, name);
+  }
+
+  /**
+   * Get the value of a shape attribute of this operation.
+   *
+   * @param name the name of the attribute
+   * @return the value of the attribute
+   */
+  public Shape getAttrShape(String name) {
+    return getAttrShape(unsafeNativeHandle, name);
+  }
+
+  /**
+   * Get the value of a shape list attribute of this operation.
+   *
+   * @param name the name of the attribute
+   * @return the value of the attribute
+   */
+  public Shape[] getAttrShapeList(String name) {
+    return getAttrShapeList(unsafeNativeHandle, name);
+  }
+
+  /**
+   * Get the metadata of an attribute of this operation.
+   *
+   * @param name the name of the attribute
+   * @return the metadata of the attribute
+   */
+  public AttributeMetadata getAttrMetadata(String name) {
+    return getAttrMetadata(unsafeNativeHandle, name);
+  }
+
+  /**
+   * Get the value of an attribute of this operation as an {@link AttrValue} proto.
+   *
+   * @param name the name of the attribute
+   * @return the value of the attribute as an {@link AttrValue} proto
+   */
+  public AttrValue getAttrValueProto(String name) {
+    return getAttrValueProto(unsafeNativeHandle, name);
+  }
```

Comment:
Just little suggestion, maybe instead of "polluting" the `GraphOperation` public namespace with all these new endpoints that are basically only used for internal purposes (by the new `Inputs` classes), we should move them all to some kind of `GraphOperationInspector` that is located somewhere in the `op.tensorflow.internal` package and add a `public GraphOperationInspector inspector()` function here?

Reply:
Where do you think I should put the demarcation though?  I could do just attributes, but then I should probably do inputs too, and then it's close enough to consumers and outputs, and then there's nothing left in `GraphOperation`.

Reply:
True, I'll leave the existing methods where they are and I think it is find as well to have `inputs` at the `GraphOperation` level since there are already public endpoints to retrieve its outputs. So let's stick to attributes then.